proto/control-api/src/control/mod.rs

```diff
@@ -0,0 +1,478 @@
+//! [`ControlApi`] definitions.
+//!
+//! [`ControlApi`]: Api
+
+// TODO: Remove once annoying false positive is fixed:
+//       https://github.com/rust-lang/rust-clippy/issues/6902
+#![allow(clippy::use_self)]
+
+pub mod endpoint;
+pub mod member;
+pub mod room;
+
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use derive_more::Display;
+
+pub use self::{endpoint::Endpoint, member::Member, room::Room};
+
+/// [Control API] used to control [Medea] server.
+///
+/// [Control API]: https://tinyurl.com/yxsqplq7
+/// [Medea]: https://git.instrumentisto.com/streaming/medea
+#[async_trait(?Send)]
+pub trait Api {
+    /// Creates a new [`Room`].
+    ///
+    /// # Errors
+    ///
+    /// With [`ErrorCode::RoomAlreadyExists`] if a [`Room`] with the provided
+    /// [`Id`] already exists.
+    ///
+    /// [`Id`]: room::Id
+    async fn create_room(&self, spec: Room) -> Result<Sids, ErrorResponse>;
```

Comment:
I'd like to keep the generalized interface (like in gRPC spec) in the core trait. For the calling side we may provide "ext-trait batteries" for `Room`, `Member` and `Endpoint`, though.

We will definitely add more media `Element`s in future, and poking every time with methods here would be unnecessary fuss.